window/src/os/x11/window.rs

```diff
@@ -872,6 +896,22 @@ impl XWindowInner {
         }
     }
 
+    fn set_text_cursor_position(&mut self, cursor: Rect) {
+        self.last_cursor_position = cursor;
+        self.update_ime_position();
+    }
+
+    fn update_ime_position(&mut self) {
+        if !self.has_focus {
+            return;
+        }
+        self.conn().ime.borrow_mut().update_pos(
+            self.window_id,
+            self.last_cursor_position.max_x() as i16,
```

Comment:
I think this might look slightly better if the x value is in the middle of the rect (eg: `(max_x() + min_x()) / 2`) as, at least on Gnome, the candidate window points to the right hand edge which looks slightly "off":

![Screenshot from 2021-08-18 08-04-16](https://user-images.githubusercontent.com/117777/129924111-f57e46fa-c811-4087-ba27-b4cf68ef919c.png)

I don't know if other IMEs look different though!

Reply:
I checked how other gui frameworks do it and consensus seems to be to set the IME position under the cursor to the left, so I changed the x position to `min_x`.